erpnext/hr/doctype/salary_structure/salary_structure.js

```diff
@@ -32,14 +32,120 @@ frappe.ui.form.on('Salary Structure', {
+
+		frm.add_custom_button(__("Add Employees"),function () {
+			frm.trigger('add_employees')
+		})
 		
-	},	
+	},
+
+	add_employees:function (frm) {
+		var emp = new frappe.ui.Dialog({
+			title: __("Add Employees"),
+			fields: [
+				{fieldname:'company', fieldtype:'Link', options: 'Company', label: __('Company')},
+				{fieldname:'branch', fieldtype:'Link', options: 'Branch', label: __('Branch')},
+				{fieldname:'department', fieldtype:'Link', options: 'Department', label: __('Department')},
+				{fieldname:'designation', fieldtype:'Link', options: 'Designation', label: __('Designation')},
+				{fieldname:'base_variable', fieldtype:'Section Break'},
+				{fieldname:'base', fieldtype:'Currency', label: __('Base')},
+				{fieldname:'base_col_br', fieldtype:'Column Break'},
+				{fieldname:'variable', fieldtype:'Currency', label: __('Variable')},
+				{fieldname:'check_sec', fieldtype:'Section Break'},
+				{fieldname:'check_all', fieldtype:'Button', label: __('Check All')},
+				{fieldname:'check_col_br', fieldtype:'Column Break'},
+				{fieldname:'uncheck_all', fieldtype:'Button', label: __('Uncheck All')},
+				{fieldname:'check_col_br2', fieldtype:'Column Break'},
+				{fieldname:'set_employee', fieldtype:'Button', label: __('Set')},
+				{fieldname:'employees_sec', fieldtype:'Section Break', label: __('Employees')},
+				{fieldname:'employees_html', fieldtype:'HTML', label: __('Employees')}
+			]
+		});
+		frm.$empDialog = emp;
+		frm.trigger('getEmployees');
+		frm.$empDialog.get_input('company').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('department').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('branch').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('designation').on('change', function () {
+			frm.trigger('getEmployees');
+		});
+		frm.$empDialog.get_input('check_all').on('click', function () {
+			frm.$empDialog.__check_all = true;
+			frm.trigger('employeeCheckToggle');
+		});
+		frm.$empDialog.get_input('uncheck_all').on('click', function () {
+			frm.$empDialog.__check_all = false;
+			frm.trigger('employeeCheckToggle');
+		});
+		frm.$empDialog.get_input('set_employee').addClass('btn-primary').on('click', function () {
+			frm.trigger('setEmployeesInChildTable');
+		});
+		frm.$empDialog.show();
+	},
+
+	getEmployees:function (frm) {
+		var filters = frm.$empDialog.get_values();
+		delete filters.check_all;
+		delete filters.uncheck_all;
+		frappe.call({
+			method:'erpnext.hr.doctype.salary_structure.salary_structure.get_employees',
+			args:{
+				filters: filters
+			},
+			callback:function (r) {
+				frm.$empDialog.__data = r.message;
+				frm.trigger('setEmployee')
+			}
+		})
+	},
+	
+	
+	setEmployeesInChildTable:function (frm) {
+		var employees = $.map(frm.doc.employees, function(d) { return d.employee });
```

Comment:
Remove the checkbox code, just add the employees in the table and then let the user decide